hpcbench/driver.py

```diff
@@ -520,17 +520,23 @@ def _check_metrics(cls, schema, metrics):
     def _check_metric(cls, schema, metric, name, value):
         if isinstance(metric, Metric):
             if not isinstance(value, metric.type):
-                message = "Unexpected type for metrics {}".format(name)
+                message = "Unexpected type for metrics {}:\n".format(name)
+                message += "expected {}, but got {}".format(metric.type,
+                                                            type(value))
                 raise Exception(message)
         elif isinstance(metric, list):
             if not isinstance(value, list):
                 message = "Unexpected type for metrics {}".format(name)
+                message += "expected {}, but got {}".format(type(list),
+                                                            type(value))
                 raise Exception(message)
             for item in value:
                 cls._check_metric(schema, metric[0], name, item)
         elif isinstance(metric, dict):
             if not isinstance(value, dict):
                 message = "Unexpected type for metrics {}".format(name)
+                message += "expected {}, but got {}".format(type(dict),
```

Comment:
Same comment here, `type` should not be used.

Reply:
yes, you're right of course.. when breaking the string into too, I did something stupid - I'll fix that.